docs/references/t-schnorr-how-it-works.mdx

```diff
@@ -0,0 +1,65 @@
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Threshold Schnorr signatures reference
+
+<MarkdownChipRow labels={["Reference"]} />
+
+## Overview
+Currently, this page (only) contains a technical specification for performing Ed25519 hierarchical key derivation.
+This specification is relevant when using the Ed25519 variant of threshold Schnorr signatures on ICP.
+It is, for example, of interest as reference for developers who implement libraries that perform this type of key derivation.
+
+## Ed25519 hierarchical key derivation
+
+This section describes a child key derivation (CKD) function for computing child public keys from parent public keys.
+The section is inspired by [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) and uses similar wording and structure.
+
+### Motivation
+
+To support the Ed25519 variant of threshold Schnorr signatures on the Internet Computer, a key derivation scheme compatible with Ed25519 signatures is required.
+For a respective signing service on the Internet Computer to be efficient, the signing subnet maintains only a single master key pair and _derives_ signing child keys for each canister.
+Although there exist various hierarchical key derivation schemes (e.g., [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki), [SLIP10](https://github.com/satoshilabs/slips/blob/master/slip-0010.md), [BIP32-Ed25519](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf), [Schnorrkel](https://github.com/w3f/schnorrkel)), none of the analyzed schemes is suitable for the Internet Computer's (threshold) setting.
+For this reason, a new derivation scheme is specified here.
+The specified scheme does not make use of _clamping_, because it is unnecessary in the given setting, and satisfies the following requirements:
+
+- Off-chain availability: New public keys can be computed off-chain from a master public key without requiring interaction with the IC.
+- Hierarchical derivation: Derived keys are organized in a tree such that from any public key it is possible to derive new child keys. This tree can have arbitrary depth. The first level is used to derive unique canister-specific keys from the master key.
+- Compatibility: The scheme is compatible with existing standards, and it is simple to implement using existing libraries.
+
+### Conventions
+
+We will assume the elliptic curve cryptography using the field and curve parameters defined by Ed25519 (see [RFC8032, Section 5.1](https://datatracker.ietf.org/doc/html/rfc8032#section-5.1)). Variables below are either:
+
+- Integers modulo the order of the curve (referred to as n).
```

Comment:
While in most literature `n` is the typical variable used to refer to the group order, Ed25519 notation usually uses `L` instead (for example RFC 8032 calls it `L`, using `n` to refer to the bitlength of the order)

Given the existence of the Ed25519 cofactor this should probably also specify that we are referring to the prime order subgroup, rather than the overall elliptic curve which has order `8*L`

Reply:
> While in most literature n is the typical variable used to refer to the group order, Ed25519 notation usually uses L instead (for example RFC 8032 calls it L, using n to refer to the bitlength of the order)

n is now renamed to L. We also had an L for HKDF: this L was renamed to N.

> Given the existence of the Ed25519 cofactor this should probably also specify that we are referring to the prime order subgroup, rather than the overall elliptic curve which has order 8*L

I adapted the sentence to "Integers modulo the order of the curve's prime order subgroup (referred to as L)."